src/introspect/introspect.ts

```diff
@@ -12,30 +12,19 @@
  * limitations under the License.
  */
 
-import { QueryResult } from '../query-result/query-result';
-import { SqlLiteral, SqlQuery, SqlTable } from '../sql';
-import { filterMap } from '../utils';
+import { Column, QueryResult } from '../query-result';
+import { SqlQuery, SqlTable } from '../sql';
+import { dedupe } from '../utils';
 
 export interface TableInfo {
   name: string;
 }
 
-export interface ColumnInfo {
-  name: string;
-  type: string;
-}
-
-interface Limit0QueryColumnIntrospectionQuery {
+interface QueryColumnIntrospectionQuery {
   query: string;
-  header?: true;
-  typesHeader?: true;
-  sqlTypesHeader?: true;
-}
-
-function guessTypeFromValue(v: any): string | undefined {
-  if (v instanceof Date) return 'TIMESTAMP';
-  if (v === true || v === false) return 'BOOLEAN';
-  return;
+  header: true;
+  typesHeader: true;
+  sqlTypesHeader: true;
```

Comment:
can we add `context: Record<string, string>` here? Even though its not added in the getQueryColumnIntrospectionQuery, it's still a valid thing to have on the QueryColumnIntrospectionQuery